python/private/python_repository.bzl

```diff
@@ -143,12 +143,11 @@ def _python_repository_impl(rctx):
         # dyld lookup errors. To fix, set the full path to the dylib as
         # it appears in the Bazel workspace as its LC_ID_DYLIB using
         # the `install_name_tool` bundled with macOS.
-        dylib = "lib/libpython{}.dylib".format(python_short_version)
-        full_dylib_path = rctx.path(dylib)
+        dylib = "libpython{}.dylib".format(python_short_version)
         repo_utils.execute_checked(
             rctx,
             op = "python_repository.FixUpDyldIdPath",
-            arguments = [repo_utils.which_checked(rctx, "install_name_tool"), "-id", full_dylib_path, dylib],
+            arguments = [repo_utils.which_checked(rctx, "install_name_tool"), "-id", "@rpath/{}".format(dylib), "lib/{}".format(dylib)],
```

Comment:
How exactly does Bazel set the rpath? Does it add every location with `cc_import`s in it to the rpath list?

In any case, this is better than my previous solution, which ultimately just traded a non-existing absolute lib path against an existing one.

Reply:
> How exactly does Bazel set the rpath? Does it add every location with cc_imports in it to the rpath list?

Yea something like this, in general it's an implementation detail but as long as it's correctly surfaced through the CC rules it's up to bazel to handle.

> In any case, this is better than my previous solution, which ultimately just traded a non-existing absolute lib path against an existing one.

Yea I guess since the path was invalid before, it's unlikely that folks were able to depend on it. But importantly the first non-existant path was stable, and this absolute one means teams won't get cache hits if they are using it.